.github/workflows/linux.yml

```diff
@@ -0,0 +1,119 @@
+name: Linux
+
+on:
+  # allows us to run workflows manually
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - develop
+      - master
+  push:
+    branches:
+      - develop
+      - master
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - cubit: 17.1.0
+            os: ubuntu-18.04
+            URL: LINUX_CUBIT_17_1_0
+            URL_SDK: LINUX_CUBIT_SDK_17_1_0
+            
+          - cubit: 2020.2
+            os: ubuntu-20.04
+            URL: LINUX_CUBIT_2020_2
+            URL_SDK: LINUX_CUBIT_SDK_2020_2
+
+    name: 'Trelis/Cubit Svalinn Plugin ${{ matrix.cubit }} Build for Linux'
+
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Environment Variables
+        run: |
+          echo "CURRENT=$(pwd)" >> $GITHUB_ENV
+          echo "SCRIPTPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "PLUGIN_ABS_PATH=$GITHUB_WORKSPACE/.." >> $GITHUB_ENV
+          echo "GHA=ON" >> $GITHUB_ENV
+          echo "FOLDER_PKG=$GITHUB_WORKSPACE/../pkg" >> $GITHUB_ENV
+          echo "SUDO=sudo " >> $GITHUB_ENV
+          if [ "${{ matrix.cubit }}" = "17.1.0" ]; then
+            echo "CUBIT_PATH=/opt/Trelis-17.1" >> $GITHUB_ENV
+            echo "CMAKE_ADDITIONAL_FLAGS=-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=0" >> $GITHUB_ENV
+          else
+            echo "CUBIT_PATH=/opt/Coreform-Cubit-2020.2" >> $GITHUB_ENV
+          fi
+          source $GITHUB_WORKSPACE/scripts/linux_share_build.sh
+          GHA=ON setup_var ${{ matrix.cubit }}
+
+ 
+      - name: Downloading packages
+        env:   
+          CUBIT_DEB_URL: ${{ secrets[matrix.URL] }}
+          CUBIT_SDK_URL: ${{ secrets[matrix.URL_SDK] }}
+        run: |
+          mkdir -p $FOLDER_PKG
+          cd $FOLDER_PKG
+          echo ${TRELIS_PKG}
+          curl -L https://uwmadison.box.com/shared/static/${CUBIT_DEB_URL}.deb --output ${TRELIS_PKG}
+          curl -L https://uwmadison.box.com/shared/static/${CUBIT_SDK_URL}.gz --output ${TRELIS_SDK_PKG}
+
+
+      - name: Apt dependencies
+        shell: bash
+        run: |
+          source $GITHUB_WORKSPACE/scripts/linux_share_build.sh
```

Comment:
These repeated sources of the `linux_share_build.sh` file are a bit of an annoyance.

It looks like there may be a way around this 

https://github.community/t/how-to-share-shell-profile-between-steps-or-how-to-use-nvm-rvm-in-steps/16398

Reply:
I'll take a look!

thx for the intel!!